Neos.Flow/Classes/Property/TypeConverter/PersistentObjectConverter.php

```diff
@@ -215,6 +221,9 @@ public function convertFrom($source, $targetType, array $convertedChildPropertie
                 );
                 throw new InvalidTargetException($exceptionMessage, 1297935345);
             }
+            if ($isCollectionType && $originalValue !== ObjectAccess::getProperty($object, $propertyName, true)) {
+                throw new TypeConverterException(sprintf('You overwrote the collection property "%s" of type "%s". This will lead to data loss. Read more about it and how to prevent this here: ADDLINKHERE', $propertyName, $targetType), 1602324196);
```

Comment:
Just wondering:
Would it make sense to only throw the exception depending on the actual ORM behavior? i.e. someone with an Flow 5.3/Neos 4.3 installation could safely update to new patchlevel versions without risking a changed behavior?

Reply:
What do you mean with "actual ORM behavior"? In any Version 2.6+, changing the instance of a collection (with orphanRemoval) will result in a `DELETE * FROM RelatedTable WHERE parentId = X` statement. So we kind of already do this.

Reply:
> In any Version 2.6+, changing the instance of a collection [...] will result in a DELETE

Yes, _that_ behavior. Or rather: skip the exception when ORM < 2.6 is used.

Doesn't that make sense?

Reply:
ah, we don't even _allow_  ORM < 2.6... sorry ignore me